arangod/Graph/Options/OneSidedEnumeratorOptions.h

```diff
@@ -33,16 +33,23 @@ namespace graph {
 
 struct OneSidedEnumeratorOptions {
  public:
-  OneSidedEnumeratorOptions(size_t minDepth, size_t maxDepth);
+  OneSidedEnumeratorOptions(size_t minDepth, size_t maxDepth, bool isDisjoint,
+                            bool isSatelliteLeader);
 
   ~OneSidedEnumeratorOptions();
 
-  size_t getMinDepth() const;
-  size_t getMaxDepth() const;
+  [[nodiscard]] size_t getMinDepth() const;
+  [[nodiscard]] size_t getMaxDepth() const;
+  [[nodiscard]] bool isDisjoint() const { return _isDisjoint; }
+  [[nodiscard]] bool isSatelliteLeader() const { return _isSatelliteLeader; }
+  // TODO [GraphRefactor]: _isDisjoint && _isSatelliteLeader only relevant
+  // together. One boolean is enough here.
```

Comment:
Does not belong here.
And: The comment is not true.
isSatelliteLeader only has a meaning in isDisjoint case, but there can be isDisjoint with satelliteLeader true and false!
So we need an enum of 3 values `NORMAL``DISJOINT_SATELLITE_LEADER``DISJOINT_SATELLITE_FOLLOWER` if we want to have a single parameter only.

Reply:
Removed wrong TODO. Refactored the API. isDisjoint and isSatelliteLeader now part of PathValidatorOptions